bin/node/cli/src/service.rs

```diff
@@ -89,6 +95,10 @@ macro_rules! new_full_start {
 				Ok(import_queue)
 			})?
 			.with_rpc_extensions(|builder| -> std::result::Result<RpcExtension, _> {
+
+				// TODO: Do something with this stream
+				let justification_stream = finality_proof_subscription.stream();
```

Comment:
It's not enough to have single stream / single subscription (well it is, but it's makes the code way more complicated). The way this work in e.g. `client` is that we have a list of subscriptions and the sender notifies all of them. Then the receiving ends can be polled independently.

Subscribing to to justifications (creating new stream / receiving end) should only happen when the respective RPC call  has been made.